lib/SILOptimizer/Mandatory/TFPartition.cpp

```diff
@@ -2865,7 +2900,7 @@ void PartitionCloner::handleSwitchEnum(SwitchEnumInst *inst) {
                 BA, loc, remapType(intFieldSILType),
                 /*isScalar*/ true, nextSendID, FP.configuration);
 
-  assert(inst->getNumCases() == 2 &&
+  assert(inst->getNumSuccessors() == 2 &&
```

Comment:
Heads-up that this restriction will be lifted in https://github.com/apple/swift/pull/17698.